pkg/cloud/amazon/s3_storage.go

```diff
@@ -335,9 +359,32 @@ func newClient(
 		opts.Config.LogLevel = aws.LogLevel(aws.LogDebugWithRequestRetries | aws.LogDebugWithRequestErrors)
 	}
 
-	sess, err := session.NewSessionWithOptions(opts)
-	if err != nil {
-		return s3Client{}, "", errors.Wrap(err, "new aws session")
+	var sess *session.Session
```

Comment:
The case where auth is using assume role requires a session to already exist. However where we originally had the switch case was before we made the session. 

I didn't see anything about the switch case that needed to run first (it was just adding opts), so instead of making a separate if check after we make the session, I thought it would be cleaner just to move the entire switch case to after we have added all the opts.